sdk/core/azure-core/inc/azure/core/http/policy.hpp

```diff
@@ -47,16 +45,15 @@ namespace Azure { namespace Core { namespace Http {
      *
      * @param context #Azure::Core::Context so that operation can be cancelled.
      * @param request An #Azure::Core::Http::Request being sent.
-     * @param policy #Azure::Core::Http::NextHttpPolicy to invoke after this policy has been
-     * applied.
+     * @param nextPolicy An iterator referencing the current called policy.
      *
      * @return An #Azure::Core::Http::RawResponse after this policy, and all subsequent HTTP
      * policies in the stack sequence of policies have been applied.
      */
     virtual std::unique_ptr<RawResponse> Send(
         Context const& context,
         Request& request,
-        NextHttpPolicy policy) const = 0;
+        std::vector<std::unique_ptr<HttpPolicy>>::const_iterator nextPolicy) const = 0;
```

Comment:
Nit: why not encapsulate? https://github.com/Azure/azure-sdk-for-cpp/pull/1734#discussion_r582551593

Reply:
Could you elaborate?  what would we like to hide?

Before this change, we were using the class NextHttpPolicy as a helper state to know the index of the current policy and the next one to be called from it.
I proposed removing this class as we can use a const iterator directly for the same goal.

Reply:
but you expose `std::vector<>` and we can never change it

Reply:
oh yes, you are right about that.  I should've though about that.
I think I can remove it use another approach.  Otherwise, I think we would just leave the NextHttpPolicy class as the way to abstract it.
Nice catch!

Reply:
it's okay. my mind was wandering.